lib/chippy_web/controllers/page_controller.ex

```diff
@@ -21,16 +21,16 @@ defmodule ChippyWeb.PageController do
     |> render("profile.html")
   end
 
-  def profile_save(conn, %{"profile" => %{"user_id" => user_id}}) do
+  def profile_save(conn, %{"profile" => %{"user_id" => user_id} = form}) do
     conn
     |> put_session(:user_id, user_id)
     |> put_flash(:info, "Profile saved successfully.")
-    |> redirect(to: Routes.page_path(conn, :index))
+    |> redirect(to: Map.get(form, "next", Routes.page_path(conn, :index)))
   end
 
   def sprint(conn, %{"sid" => sprint_id}) do
     case get_session(conn, :user_id) do
-      user_id when user_id != nil ->
+      user_id when user_id not in [nil, ""] ->
```

Comment:
This improvement catches empty usernames, but not usernames which are all spaces. Maybe a future improvement to validate usernames further?

Reply:
Whoops, fixed this in the wrong PR, but hopefully it works. 🙂